cmd/minishift/cmd/start_preflight.go

```diff
@@ -394,38 +394,26 @@ func checkLibvirtDefaultNetworkActive() bool {
 
 // checkHypervDriverSwitch returns true if Virtual Switch has been selected
 func checkHypervDriverSwitch() bool {
-	switchEnvName := "HYPERV_VIRTUAL_SWITCH"
 	posh := powershell.New()
 	defer posh.Close()
 
-	switchEnvValue := os.Getenv(switchEnvName)
+	switchName := viper.GetString(configCmd.HypervVirtualSwitch.Name)
 
-	if switchEnvValue == "" {
-		checkIfDefaultSwitchExists := `Get-VMSwitch -Id c08cb7b8-9b3c-408e-8e30-5e16a3aeb444 | ForEach-Object { $_.SwitchType }`
-		fmt.Printf("\n   'Default Switch' ... ")
-		stdOut, _ := posh.Execute(checkIfDefaultSwitchExists)
-		if strings.Contains(stdOut, "Internal") {
+	// check for default switch
+	if switchName == "" {
+		checkIfDefaultSwitchExists := `Get-VMSwitch -Id c08cb7b8-9b3c-408e-8e30-5e16a3aeb444 | ForEach-Object { $_.Name }`
```

Comment:
@gbraad This ID `c08cb7b8-9b3c-408e-8e30-5e16a3aeb444` is not going to differ on different windows systems?

Reply:
It will not, but the Id should not be passed as a string, but as an uuid object. @gbraad you mentioned this to me on IRC once... ??

Reply:
it is const... this is a static identifier. might be able to add this as a `constants.go` as it makes it more visible.

however, this is also unnamed in the libmachine code...

Reply:
https://blogs.technet.microsoft.com/virtualization/2017/11/13/whats-new-in-hyper-v-for-windows-10-fall-creators-update/

Reply:
addressed...

Reply:
@gbraad What i meant was we should form a guid object which will be passed to the the command `Get-VMSwitch -Id` it takes a guid object.. here the optional parameters section if you see the `Id` flag it lists type as `Guid[]`.

We can run this in posh to form the guid ...
`$id=[guid]"c08cb7b8-9b3c-408e-8e30-5e16a3aeb444"` and use `$id` in the command...

Reply:
is this really necessary? PowerShell already handles the guid type conversion... if the value is NOT a guid, it will fail... you can try this by leaving a character off

Reply:
> PowerShell already handles the guid type conversion...

Than its Okay..